src/core_plugins/kibana/public/dashboard/top_nav/clone_modal.js

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  KuiModal,
+  KuiModalHeader,
+  KuiModalHeaderTitle,
+  KuiModalBody,
+  KuiModalBodyText,
+  KuiModalFooter,
+  KuiButton
+} from 'ui_framework/components';
+
+export class DashboardCloneModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      newDashboardName: props.title
+    };
+  }
+
+  componentDidMount() {
+    this.confirmButton.focus();
+  }
+
+  cloneDashboard = () => {
+    this.props.onClone(this.state.newDashboardName);
+  };
+
+  onInputChange = (event) => {
+    this.setState({ newDashboardName: event.target.value });
+  };
+
+  onKeyDown = (event) => {
+    if (event.keyCode === 27) {
+      this.props.onClose();
+    }
+  };
+
+  render() {
+    return (
+      <KuiModal
+        data-tests-subj="dashboardCloneModal"
+        aria-label="Clone a dashboard"
+        className="dashboardCloneModal"
+        onKeyDown={ this.onKeyDown }
+      >
+        <KuiModalHeader>
+          <KuiModalHeaderTitle>
+            Clone Dashboard
+          </KuiModalHeaderTitle>
+        </KuiModalHeader>
+        <KuiModalBody>
+          <KuiModalBodyText className="kuiVerticalRhythm">
+            Please enter a new name for your dashboard.
+          </KuiModalBodyText>
+          <KuiModalBodyText className="kuiVerticalRhythm">
+            <input
+              data-test-subj="clonedDashboardTitle"
+              className="kuiTextInput kuiTextInput--large"
+              value={ this.state.newDashboardName }
+              onChange={ this.onInputChange } />
+          </KuiModalBodyText>
```

Comment:
Should the model also let the user update the dashboard description for the clone? I could see a user cloning a dashboard and wanting to describe what the clone is going to be used for. Might be nice to complete this action in a single screen.